envi/tests/test_arch_arm.py

```diff
@@ -1732,66 +2095,95 @@ def test_envi_arm_assorted_instrs(self):
         goodcount = 0
         goodemu = 0
         bademu = 0
-        for archz, bytez, va, reprOp, iflags, emutests in instrs:
+        for tidx, testinfo in enumerate(instrs):
+            archz, bytez, va, reprOp, iflags, emutests = testinfo
             ranAlready = False  # support for run once only
-            #itterate through architectures 
+            #iterate through architectures 
             for key in ARCH_REVS:
                 test_arch = ARCH_REVS[key]
                 if ((not ranAlready) or (not self.armTestOnce)) and ((archz & test_arch & self.armTestVersion) != 0):
-                    ranAlready = True
-                    op = vw.arch.archParseOpcode(binascii.unhexlify(bytez), 0, va)
-                    redoprepr = repr(op).replace(' ','').lower()
-                    redgoodop = reprOp.replace(' ','').lower()
-                    if redoprepr != redgoodop:
-                        num, = struct.unpack("<I", binascii.unhexlify(bytez))
-                        bs = bin(num)[2:].zfill(32)
-                        badcount += 1
-                        raise Exception("%d FAILED to decode instr:  %.8x %s - should be: %s  - is: %s" % \
-                                (goodcount, va, bytez, reprOp, repr(op) ) )
-                        # self.assertEqual((goodcount, bytez, redoprepr), (goodcount, bytez, redgoodop))
+                    try:
+                        logger.info("%s: %r", tidx, testinfo)
```

Comment:
Should this be outside of the `for key in ARCH_REVS` bit, or also include the `key` variable?

Reply:
key was never intended as a data point here, but intended to allow for different architecture configurations to be have a different set of tests run.  not the important if you'd like to rip that out and just run them all.